### PR TITLE
Fix CI after GHA's drop of node16 actions

### PR DESCRIPTION
**Pull request by Quinn-With-Two-Ns**
Build the test server in an ubuntu 18 container

### CODE REVIEW

File: .github/workflows/prepare-release.yml [hunk 1]
```diff
@@ -24,13 +24,19 @@ on:
...
+        required: true
+        default: "true"
 env:
   INPUT_REF: ${{ github.event.inputs.ref }}
   INPUT_TAG: ${{ github.event.inputs.tag }}
 
 jobs:
   create_draft_release:
     name: Create Github draft release
+    if: github.event.inputs.do_create_release == 'true'
```
Comment (Quinn-With-Two-Ns) [inline .github/workflows/prepare-release.yml:39]: This is just for testing and will be removed

File: .github/workflows/prepare-release.yml [hunk 2]
```diff
@@ -92,7 +98,7 @@ jobs:
           distribution: "temurin"
 
       - name: Set up Gradle
-        uses: gradle/actions/setup-gradle@v3
+        uses: gradle/actions/setup-gradle@v4
```
Comment (mjameswh) [inline .github/workflows/prepare-release.yml:101]: There are other places we use `gradle/actions/setup-gradle@v3`. Should update those too.

File: .github/workflows/prepare-release.yml [hunk 3]
```diff
@@ -138,43 +144,45 @@ jobs:
...
+        run: |
+          ./gradlew :temporal-test-server:build
 
+      - name: Build native test server (Docker)
+        if: matrix.os_family == 'Linux'
+        run: |
+          docker run \
+            --rm -w /github/workspace -v "$(pwd):/github/workspace" \
+            $(docker build -q ./docker/native-image) \
+            sh -c "./gradlew :temporal-test-server:build"
```
Comment (mjameswh) [inline .github/workflows/prepare-release.yml:181]: Now that you moved installation stuff to a docker image, you probably no longer need the `sh -c` and quotes.

File: .github/workflows/prepare-release.yml (hunk 2 above)
Comment (Quinn-With-Two-Ns) [reply .github/workflows/prepare-release.yml:95]: Done

File: .github/workflows/prepare-release.yml (hunk 3 above)
Comment (Quinn-With-Two-Ns) [reply .github/workflows/prepare-release.yml:175]: Yep, fixed

File: docker/native-image/dockerfile [hunk 4]
```diff
@@ -0,0 +1,8 @@
+FROM ubuntu:18.04
```
[the remark this answers is not in the document]
Comment (Quinn-With-Two-Ns) [reply docker/native-image/dockerfile:3]: Added a comment

**Review by mjameswh — APPROVED** [review]
Thanks for completing that.

**Pull request merged**